### DISCUSSION TITLE
Ethernet Static IP

### DISCUSSION THREAD

**Discussion by kgibso01** [xblax/flashforge_ad5m_klipper_mod#117] [Q&A] [answered]
Thanks for this mod!

A bit of a 3D printing noob but always been a hack/fiddler, had been learning about klipper and have installed this mod. Got it all going and it seems to be working great. 

One thing I couldn't figure out is how to set a static ip? I saw the spot about adding custom config files through the klipper_mod folder but can't seem to get it to work.

Does anyone have a network config for static ip ethernet port they wouldn't mind sharing?

**Comment by kgibso01** [comment]
I went about this differently and set a static DHCP reservation in my pi-hole router configs and it is working now on the correct IP. 

I never did figure out how to set it through the interfaces config file but, same result so good to go.

**Comment by THEGxHOxST** [comment, marked as answer]
I'm glad you found a workaround!
But for anyone else.

nano /etc/network/interfaces

replace 

auto eth0
iface eth0 inet dhcp
  pre-up /etc/network/nfs_check
  wait-delay 15
  hostname $(hostname)

With

auto eth0
  iface eth0 inet static
  address 192.168.1.100(Enter desired ip here)
  netmask 255.255.255.0
  network 192.168.1.0
  broadcast 192.168.1.255
  gateway 192.168.1.1
  pre-up /etc/network/nfs_check
  wait-delay 15
  hostname $(hostname)

**Comment by kgibso01** [comment]
Thanks. I may go back around and do it this way as well. Feels more proper. Lol.
